core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/LoginViewModel.kt

```diff
@@ -0,0 +1,17 @@
+package com.ustadmobile.core.viewmodel
+
+data class LoginUiState(
+    val username: String = "",
+    val password: String = "",
+    val usernameEnabled: Boolean = true,
+    val passwordEnabled: Boolean = true,
+    val isEmptyPassword: Boolean = false,
```

Comment:
I know the field was on the view interface, but please remove the passwordEmpty and isEmptyUsername. Just keep usernameError and passwordError.